packages/signals/types/tests/produce-actions.lint.ts

```diff
@@ -0,0 +1,41 @@
+/// <reference types="@types/jest" />
+
+import { produceActions } from '../../';
+
+describe('[TEST]: produceActions', () => {
+  it('should error on invalid use cases', () => {
+    class ActionWithoutType {
+      constructor(readonly event: MessageEvent) {}
+    }
+
+    produceActions(); // $ExpectError
+    produceActions({}); // $ExpectError
+
+    produceActions({
+      action: ActionWithoutType // $ExpectError
+    });
+
+    produceActions({
+      action: { // $ExpectError
+        type: 'Plain Object'
+      }
+    });
+  });
+
+  it('should infer correct return types', () => {
+    class ValidAction {
+      static readonly type = 'Valid Action';
+
+      constructor(readonly name: string) {}
+    }
+  
+    class ValidActionWithMultipleParameters {
+      static readonly type = 'Valid Action';
+
+      constructor(readonly name_1: string, readonly name_2: number) {}
+    }
+
+    produceActions({ action: ValidAction }); // $ExpectType { action: (name: string) => Observable<void>; }
```

Comment:
What do you think about this action dispatching function returning a promise as opposed to an observable?

Reply:
Is it to wrap it with `createDispatchFn().toPromise()` or smth?

Reply:
Yes. Because it makes the consumption of the result a lot simpler. There will only ever be one result. This is what promises are good at. What do you think?

Reply:
I’m not against promises, my only concern is the cancellation.

Reply:
hmmm... good point.